django/views/debug.py

```diff
@@ -347,12 +347,18 @@ def get_traceback_data(self):
             self.template_does_not_exist = True
             self.postmortem = self.exc_value.chain or [self.exc_value]
 
+        # Limits the size of returning string to reduce memory usage
+        repr_instance = reprlib.Repr()
+        repr_instance.maxstring
+        repr_instance.maxlist
+        repr_instance.maxdict
+
         frames = self.get_traceback_frames()
         for i, frame in enumerate(frames):
             if "vars" in frame:
                 frame_vars = []
                 for k, v in frame["vars"]:
-                    v = pprint(v)
+                    v = repr_instance.repr(v)
                     # Trim large blobs of data
                     if len(v) > 4096:
                         v = "%s… <trimmed %d bytes string>" % (v[0:4096], len(v))
```

Comment:
This message will need adjusting as it will be trimming from the arbitrary length provided by reprlib, not the true length. Or perhaps this branch can be removed as reprlib will do trimming for us, replacing with "...".